lint/rule.go

```diff
@@ -372,6 +372,28 @@ func extractDisabledRuleName(commentLine string) string {
 // ruleIsEnabled returns true if the rule is enabled (not disabled by the comments
 // for the given descriptor or its file), false otherwise.
 func ruleIsEnabled(rule ProtoRule, d desc.Descriptor, aliasMap map[string]string) bool {
+	// Sanity check: All rules are disabled on a deprecated descriptor.
+	deprecated := false
+	switch v := d.(type) {
+	case *desc.EnumDescriptor:
+		deprecated = v.GetEnumOptions().GetDeprecated()
+	case *desc.EnumValueDescriptor:
+		deprecated = v.GetEnumValueOptions().GetDeprecated()
+	case *desc.FieldDescriptor:
+		deprecated = v.GetFieldOptions().GetDeprecated()
+	case *desc.FileDescriptor:
+		deprecated = v.GetFileOptions().GetDeprecated()
+	case *desc.MessageDescriptor:
+		deprecated = v.GetMessageOptions().GetDeprecated()
+	case *desc.MethodDescriptor:
+		deprecated = v.GetMethodOptions().GetDeprecated()
+	case *desc.ServiceDescriptor:
+		deprecated = v.GetServiceOptions().GetDeprecated()
+	}
+	if deprecated {
+		return false
+	}
+
```

Comment:
Can you create a function for it? And we can return in the `case`, like:
```golang
func isDescriptorDeprecated(d desc.Descriptor) bool {
  switch v := d.(type) {
  case *desc.EnumDescriptor:
    return v.GetEnumOptions().GetDeprecated()
  case *desc.EnumValueDescriptor:
    return v.GetEnumValueOptions().GetDeprecated()
  }
  return false
}
```

Reply:
I am going to add it to [protoreflect instead](https://github.com/jhump/protoreflect/issues/300), and I will back this out once that lands.

Reply:
Good idea! Thanks!

Reply:
Can we merge this in the meantime? :)

Reply:
Approved